src/levanter/infra/ray_tpu.py

```diff
@@ -222,44 +193,140 @@ def _multislice_info_to_env_vars(multislice: MultisliceInfo) -> dict[str, str]:
             try:
-                logger.exception(f"Failed to create placement group {pg_name}. Removing it. Error: {e}")
-                ray.util.remove_placement_group(pg)  # don’t leave a half-ready PG behind
-            except Exception as remove_pg_exception:
-                logger.exception(f"Failed to remove placement group {pg_name} after failure: {remove_pg_exception}")
-            raise RuntimeError(f"Failed to create placement group {pg_name}") from e
+                if ray.get(health, timeout=_HEALTH_CHECK_TIMEOUT):
+                    healthy_members.append(member)
+                else:
+                    logger.warning(f"{self.get_actor_pool_name()} actor pool member {self.get_actor_name_from_actor_info(member.actor_info)} is unhealthy. Removing from actor pool.")
+                    unhealthy_members.append(member)
+            except (RayActorError, RayTaskError, ActorDiedError, ActorUnavailableError, GetTimeoutError) as e:
+                logger.warning(f"{self.get_actor_pool_name()} actor pool member {self.get_actor_name_from_actor_info(member.actor_info)} is dead or unavailable. Removing from actor pool. Error: {e}")
+                unhealthy_members.append(member)
+
+        # NOTE: For simplicity, we serially process the unhealthy actors, rather than doing it in parallel.
+        for unhealthy_member in unhealthy_members:
+            # This is a synchronous blocking call.
+            _stop_actor(unhealthy_member.actor)
+
+        self._actor_pool = healthy_members
+        logger.info(f"{self.get_actor_pool_name()} actor pool members after unhealthy members: {[self.get_actor_name_from_actor_info(member.actor_info) for member in self._actor_pool]}")
+
+    def _add_members_to_actor_pool(self, desired_num_actors: int) -> None:
+        logger.info(f"{self.get_actor_pool_name()} actor pool members before adding members: {[self.get_actor_name_from_actor_info(member.actor_info) for member in self._actor_pool]}")
+        if len(self._actor_pool) >= desired_num_actors:
+            logger.info(f"{self.get_actor_pool_name()} actor pool has {len(self._actor_pool)} members, and we wanted {desired_num_actors}. Skipping adding members.")
+            return
+
+        # if we don't have enough slices, create more
+        logger.info(f"{self.get_actor_pool_name()} actor pool has {len(self._actor_pool)} members, but we want {desired_num_actors}. Creating more.")
+
+        actors = [self.create_actor() for _ in range(desired_num_actors - len(self._actor_pool))]
+
+        actors_and_actor_info_awaitables = [(actor, self.get_actor_info_future(actor)) for actor in actors]
+        logger.info(f"{self.get_actor_pool_name()} actor pool waiting for {len(actors)} new actors to start...")
+        for actor, actor_info_awaitable in actors_and_actor_info_awaitables:
+            try:
+                actor_info: ActorInfoT = ray.get(actor_info_awaitable, timeout=_START_ACTOR_TIMEOUT)  # TODO: make this overridable
```

Comment:
i think we're ok with this just taking forever fwiw. Like literally days. We use Ray as a janky batch submission thing and it's fine if there's no capacity for a while

Reply:
Changed this to 7 days.